tests/unit/src/test_conditionsStore.cpp

```diff
@@ -708,8 +706,66 @@ SCENARIO( "Providing multiple derived conditions", "[ConditionStore][DerivedMult
 				REQUIRE( mockProvPrefixShips.values.size() == 3 );
 			}
 		}
+		WHEN( "adding a prefixed provider that is in the subset of the prefixed provider" )
+		{
+			auto mockProvPrefixShipsA = MockConditionsProvider();
+			mockProvPrefixShipsA.SetRWPrefixProvider(store, "ships: A:");
+			store["ships: A: something"] = 40;
+			THEN( "only the initial provider is used" )
+			{
+				REQUIRE( mockProvPrefixShipsA.values["ships: A: something"] == 0 );
+				REQUIRE( mockProvPrefixShips.values["ships: A: something"] == 40 );
+			}
+		}
+		WHEN( "adding a named provider that is in the subset of the prefixed provider" )
+		{
+			auto mockProvPrefixShipsA = MockConditionsProvider();
+			mockProvPrefixShipsA.SetRWNamedProvider(store, "ships: A:");
+			store["ships: A: something"] = 40;
+			THEN( "only the initial provider is used" )
+			{
+				REQUIRE( mockProvPrefixShipsA.values["ships: A: something"] == 0 );
+				REQUIRE( mockProvPrefixShips.values["ships: A: something"] == 40 );
+			}
+		}
+		WHEN( "adding a prefixed provider that is in the superset of the prefixed provider" )
+		{
+			auto mockProvPrefixShi = MockConditionsProvider();
+			mockProvPrefixShi.SetRWPrefixProvider(store, "shi");
+			store["ships: A: something"] = 40;
+			THEN( "only the superset provider is used" )
+			{
+				REQUIRE( mockProvPrefixShi.values["ships: A: something"] == 40 );
+				REQUIRE( mockProvPrefixShips.values["ships: A: something"] == 0 );
+			}
+		}
+	}
+	GIVEN( "A pre-existing condition in a store" )
+	{
+		auto store = ConditionsStore();
+		store["ships: A"] = 40;
+		WHEN(" a prefixed provider gets added which has the condition in range")
+		{
+			auto mockProvPrefixShips = MockConditionsProvider();
+			mockProvPrefixShips.SetRWPrefixProvider(store, "ships: ");
+			THEN( "The condition is lost/removed" )
+			{
+				REQUIRE( store["ships: A"] == 0 );
+				REQUIRE( mockProvPrefixShips.values["ships: A"] == 0);
+			}
+			THEN( "Adding a sub-prefix-condition should not be possible")
+			{
+				auto mockProvPrefixShipsLarge = MockConditionsProvider();
```

Comment:
This variable seems unnecessary since it is never set.

Reply:
Good catch! I was using the wrong provider on line 759.

Reply:
Ah I thought that was intentional because of the string.

Reply:
The idea is that the provider for "ships: Large: " should just not be registered at all, because a provider for "ships: " already exists.